Set default log formatter when `--verify` is set

qiuzhong opened the pull request:
When no log formatter is set, `./wpt run --verify` is not useful as
no output is printed. This patch tries to set the default log formatter
to `log_mach` with the value of sys.stdout.
Fixed: #15013

qiuzhong committed:
Set default log formatter when `--verify` is set

When no log formatter is set, `./wpt run --verify` is not useful as
no output is printed. This patch tries to set the default log formatter
to `log_mach` with the value of sys.stdout.
Fixed: #15013

jugglinmike commented on tools/wptrunner/wptrunner/wptlogging.py:
```diff
@@ -28,6 +29,22 @@ def setup_stdlib_logger():
+    set a default log formatter to mach and its value is sys.stdout.
+    This is useful for ./wpt run --verify."""
+    log_formatters = ("raw", "unittest", "xunit", "html", "mach", "tbpl",
+                      "grouped", "errorsummary")
```
[Also note `--log-wptreport` and `--log-wptscreenshot`.](https://github.com/web-platform-tests/wpt/blob/26c03768165078f56415db91c78ff64bcf9cc142/tools/wptrunner/wptrunner/wptcommandline.py#L329-L330) Maintained here, a list of formatters here is bound to go stale; do you think you could avoid it by inspecting the `commandline` module of the `mozlog` package? It's a singleton, so I'm not sure where would be the most appropriate place to import it.

qiuzhong replied:
I didn't notice the `--log-wptreport` and `--log-wptscreenshot`. The `log_formatters` is from the `commandline` module in `mozlog` and this is the initial value. Even so, I realize it's not a good idea to hard-code here as it might be updated at other places. I'll avoid by importing `commandline` from `mozlog` package and using `commandline.log_formatters`.

jugglinmike commented on tools/wptrunner/wptrunner/wptlogging.py:
```diff
@@ -28,6 +29,22 @@ def setup_stdlib_logger():
+        for log_formatter in log_formatters:
+            if kwargs.get("log_" + log_formatter):
+                is_formatter_set = True
+                return
```
The early `return` makes the flag unnecessary

qiuzhong replied:
I'll remove the unnecessary line.

jugglinmike reviewed (commented):
This seems more like a feature of the stability checker than of the logger. Relocating it in the stability checking code (which [already contains similar command-line argument manipulation](https://github.com/web-platform-tests/wpt/blob/26c03768165078f56415db91c78ff64bcf9cc142/tools/wptrunner/wptrunner/wptrunner.py#L313-L329)) will avoid adding complexity to this more widely-used codepath.

qiuzhong commented:
@jugglinmike , the original issue is that there is no extra output for`./wpt run --verify` command. I guess it's related to the logger configuration. `check_stability` function actually runs the test for several times and checks the consistency. Meanwhile, it's executed after the logger has been set up. I thought I should put it before `wptlogging.setup()`.

qiuzhong committed:
Remove hard-coded log formatters list and unnecessary line

Improve the code by removing the hard-coded log_formatters list in
`mozlog.commandline` and unnecessary line.

jugglinmike commented:
@jgraham you're much more familiar with the organization of wptrunner than I. Does this look like the right place to implement the desired functionality?
